tests/common/utils.ts

```diff
@@ -0,0 +1,180 @@
+
+    console.log(`${proofType} Executing transaction...`);
+    const verifier = session.verify()[proofType]();
+    const verify = withAttestation ? verifier.waitForPublishedAttestation() : verifier;
+
+    const { events, transactionResult } = await verify.execute(proof, publicSignals, vk);
+
+    console.log(`${proofType} Setting up event listeners...`);
+    const eventResults = withAttestation
+        ? handleEventsWithAttestation(events, proofType, 'verify')
+        : handleCommonEvents(events, proofType, 'verify');
+
+    console.log(`${proofType} Transaction result received. Validating...`);
+
+    const transactionInfo: VerifyTransactionInfo = await transactionResult;
+    validateVerifyTransactionInfo(transactionInfo, proofType, withAttestation);
+    validateEventResults(eventResults, withAttestation);
+
+    if (validatePoe) {
+        await validatePoE(session, transactionInfo.attestationId!, transactionInfo.leafDigest!);
+    }
+
+    console.log(`${proofType} Closing session...`);
+    await session.close();
+
+    return { eventResults, transactionInfo };
+};
+
+
+export const performVKRegistrationAndVerification = async (
+    proofType: ProofType,
+    proof: any,
+    publicSignals: any,
+    vk: string
+): Promise<void> => {
+    const session = await zkVerifySession.start().Testnet().withAccount(process.env.SEED_PHRASE!);
+
+    console.log(`${proofType} Executing VK registration and setting up event listeners...`);
+    const { events: registerEvents, transactionResult: registerTransactionResult } = await session.registerVerificationKey()[proofType]().execute(vk);
+
+    const registerResults = handleCommonEvents(registerEvents, proofType, 'vkRegistration');
+
+    const vkTransactionInfo: VKRegistrationTransactionInfo = await registerTransactionResult;
+    validateVKRegistrationTransactionInfo(vkTransactionInfo, proofType);
+    validateEventResults(registerResults, false);
+
+    console.log(`${proofType} Executing verification using registered VK and setting up event listeners...`);
+    const { events: verifyEvents, transactionResult: verifyTransactionResult } = await session.verify()[proofType]().withRegisteredVk().execute(proof, publicSignals, vkTransactionInfo.statementHash!);
+    const verifyResults = handleCommonEvents(verifyEvents, proofType, 'verify');
+
+    const verifyTransactionInfo: VerifyTransactionInfo = await verifyTransactionResult;
+    validateVerifyTransactionInfo(verifyTransactionInfo, proofType, false);
+    validateEventResults(verifyResults, false);
+
+    console.log(`${proofType} Closing session...`);
+    await session.close();
+};
+
+
+export const validateTransactionInfo = (
+    transactionInfo: TransactionInfo,
+    expectedProofType: string
+): void => {
+    expect(transactionInfo).toBeDefined();
+    expect(transactionInfo.blockHash).not.toBeNull();
+    expect(transactionInfo.proofType).toBe(expectedProofType);
+    expect(transactionInfo.status).toBe(TransactionStatus.Finalized);
```

Comment:
Perhaps it could be worth to move this check up

Reply:
Up to where - this is called as the first call in both:

`export const validateVerifyTransactionInfo = (
    transactionInfo: VerifyTransactionInfo,
    expectedProofType: string,
    expectAttestation: boolean
): void => {
    validateTransactionInfo(transactionInfo, expectedProofType);`
    
 And:
 
`export const validateVKRegistrationTransactionInfo = (
    transactionInfo: VKRegistrationTransactionInfo,
    expectedProofType: string
): void => {
    validateTransactionInfo(transactionInfo, expectedProofType);`
    
Or am i misunderstanding?